### PR TITLE
Get related records (API v2)

### PR DESCRIPTION
**Pull request by Mexidense**
Hi @tristanjahier !!!
I've implemented this API in order to getting Related Record lists. I've tried to following your client's schema. I just want to contribute to your package, only if you want or see it ok... 

Zoho API doc is here: https://www.zoho.com/crm/developer/docs/api/get-related-records.html
https://www.zohoapis.com/crm/v2/{module_api_name}/{record_id}/{related_list_api_name}

For instance, I needed the stage history for one potential: the API endpoint has this way: https://www.zohoapis.com/crm/v2/Potentials/{record_id}/Stage_History

I've developed that for using in the client like:
```bash
use Zoho\Crm\V2\Client;
$client = new Client(CLIENT_ID, CLIENT_SECRET, CLIENT_REFRESH_TOKEN, CLIENT_ACCESS_TOKEN);
$potentialId = POTENTIAL_ID;
$query = $client->related_records->potentials->getRelatedRecords($potentialId, 'Stage_History');
```

What do you think about it?

Regards!

Salva

### CODE REVIEW

**Comment by tristanjahier** [comment]
Hi @Mexidense.
Thank you, I appreciate your willingness to contribute to this package!

I have some remarks: you are right, "related records" appears as a sub-API in Zoho's documentation. But, as related records are strongly related to... records (as the name suggests), I think it would be easier and better to simply include these helpers in the "Records" sub-API.

In my opinion, the query would look better like this:
```php
$query = $client->records->potentials->relationsOf($potentialId, 'Stage_History');
```

This would simply consist in adding a `ListRelatedQuery` along with the other in the `Zoho\Crm\V2\Records` namespace. Then adding a `relationsOf()` helper method in `Zoho\Crm\V2\Records\Module` to use it.

What do you think?

Similarly, we could create queries to handle "Update" and "Delink" operations.

**Comment by tristanjahier** [comment]
Don't hesitate to make a lot of changes in the same pull request, as I will squash everything in a single commit in the end, so it does not matter how many backs and forths we have to do.

**Review by tristanjahier — CHANGES_REQUESTED** [review]
Hi @Mexidense.
Thank you, I appreciate your willingness to contribute to this package!

I have some remarks: you are right, "related records" appears as a sub-API in Zoho's documentation. But, as related records are strongly related to... records (as the name suggests), I think it would be easier and better to simply include these helpers in the "Records" sub-API.

In my opinion, the query would look better like this:
```php
$query = $client->records->potentials->relationsOf($potentialId, 'Stage_History');
```

This would simply consist in adding a `ListRelatedQuery` along with the others in the `Zoho\Crm\V2\Records` namespace. Then adding a `relationsOf()` helper method in `Zoho\Crm\V2\Records\Module` to use it.

What do you think?

Similarly, we could create queries to handle "Update" and "Delink" operations but I think we should stick to only one type of query for this PR.

Don't hesitate to make a lot of changes in the same pull request, as I will squash everything in a single commit in the end, so it does not matter how many backs and forths we have to do.

**Comment by Mexidense** [comment]
> I have some remarks: you are right, "related records" appears as a sub-API in Zoho's documentation. But, as related records are strongly related to... records (as the name suggests), I think it would be easier and better to simply include these helpers in the "Records" sub-API.

hahaha, yep, indeed, I was spending for 30 minutes thinking of if I have to create a new sub API or add at Record SubAPI... and finally, I chose this because I didn't want to modify any part or Record code but I think same like you, it's better in this way :ok_hand: 
> ```
> $query = $client->records->potentials->relationsOf($potentialId, 'Stage_History');
> ```
> 
> This would simply consist in adding a `ListRelatedQuery` along with the others in the `Zoho\Crm\V2\Records` namespace. Then adding a `relationsOf()` helper method in `Zoho\Crm\V2\Records\Module` to use it.
> 
> What do you think?

Yep, it's easier to implement in this way and I think in this way in more maintainable, less code, fewer classes, fewer worries!! 

> Don't hesitate to make a lot of changes in the same pull request, as I will squash everything in a single commit in the end, so it does not matter how many backs and forths we have to do.

D'accord mon ami!! I'll do it and send you a new review request! :+1: :+1: :raised_hands:

**Comment by tristanjahier** [comment]
> Yep, it's easier to implement in this way and I think in this way in **more maintainable, less code, fewer classes, fewer worries!!**

Yes, exactly my point of view. 👍

I am still wondering if the class should be named `ListRelatedQuery` or `ListRelationsQuery` tho... `ListRelatedQuery` at least sounds like `ListDeletedQuery`. And in the documentation there are more references to "related records" than to "relations". What do you think?

**Comment by Mexidense** [comment]
IMO, I think it's better named it as ListRelatedQuery than ListRelationsQuery because the last one is like a get the whole of related record lists (Notes, Related records, etc...)  

Another alternative could be a ListRelatedRecordQuery.. thinking of ListUpdatedRelatedRecordQuery and ListDelinkRelatedRecordQuery... but I don't know... maybe those names are so long...

 🤷‍♂️ 🤷‍♂️ 🤷‍♂️  😬 😬

**Comment by tristanjahier** [comment]
The "records" part is implied by the namespace and IMO it suffices.

**Comment by Mexidense** [comment]
Absolutely!

Ok! then, could I change it and implementing that in `ListRelatedQuery`? and remove `RelatedRecord` folder?

**Comment by tristanjahier** [comment]
Yes this is what I meant.

**Comment by Mexidense** [comment]
Ok, I'll make new update and ping you for new review! Thanks!

**Comment by Mexidense** [comment]
Hi @tristanjahier!

I've implemented new changes such as we talked, takes a look, please.

I've tested it in PSYSH:
$response = $client->records->potentials->relationsOf('RECORD_ID', 'Stage_History')->get();

Thanks!

Salva

File: src/V2/Records/ListRelatedQuery.php
Comment (tristanjahier) [inline src/V2/Records/ListRelatedQuery.php]: You should not have removed the doc here.

File: src/V2/Records/ListRelatedQuery.php
Comment (tristanjahier) [inline src/V2/Records/ListRelatedQuery.php]: I don't think that this method is very useful. I would prefer you to remove it and use the 2 separate methods that you defined.

**Review by tristanjahier — CHANGES_REQUESTED** [review]
Thank you @Mexidense. We are getting much closer to approval now. :)
I still have a few remarks tho, but most of the requested changes are simple details.

File: src/V2/Records/ListRelatedQuery.php
Comment (tristanjahier) [inline src/V2/Records/ListRelatedQuery.php]: I know that Zoho describes it as a "related list" but I don't think it is very meaningful. I would prefer to call it the "related module name" because that is what it is. Let's just call this property `$relatedModule`, and its description "The related module API name" please.
Also please update the related methods names.

File: src/V2/Records/ListRelatedQuery.php
Comment (tristanjahier) [inline src/V2/Records/ListRelatedQuery.php]: `getRelatedModule()`

File: src/V2/Records/ListRelatedQuery.php
Comment (tristanjahier) [inline src/V2/Records/ListRelatedQuery.php]: `setRelatedModule()`

File: src/V2/Records/ListRelatedQuery.php
Comment (tristanjahier) [inline src/V2/Records/ListRelatedQuery.php]: There should be a empty line above.

File: src/V2/Records/ListRelatedQuery.php
Comment (tristanjahier) [inline src/V2/Records/ListRelatedQuery.php]: "record **ID**"

File: src/V2/Records/ListRelatedQuery.php
Comment (tristanjahier) [inline src/V2/Records/ListRelatedQuery.php]: "The record **ID**"

File: src/V2/Records/ListRelatedQuery.php
Comment (Mexidense) [reply src/V2/Records/ListRelatedQuery.php]: mmm... but that class doesn't exist right now...

File: src/V2/Records/ListRelatedQuery.php
Comment (Mexidense) [reply src/V2/Records/ListRelatedQuery.php]: okkk you're right, it's better

File: src/V2/Records/ListRelatedQuery.php
Comment (Mexidense) [reply src/V2/Records/ListRelatedQuery.php]: function removed and php block removed :+1:

**Review by Mexidense — COMMENTED** [review]
Changes done @tristanjahier ! :+1:

File: src/V2/Records/ListRelatedQuery.php
Comment (tristanjahier) [reply src/V2/Records/ListRelatedQuery.php]: This file no, but the renamed one yes (src/V2/Records/ListRelatedQuery.php).

**Review by tristanjahier — APPROVED** [review]
Nice work @Mexidense, this is what I expected. I'll make a couple of changes myself then merge it.

**Comment by tristanjahier** [comment]
Could you check the box "Allow edits from maintainers" in this pull request as described [in this guide](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) please?

I would like to push commit inside the PR before squashing & merging. 😉

**Comment by Mexidense** [comment]
> Could you check the box "Allow edits from maintainers" in this pull request as described [in this guide](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) please?
> 
> I would like to push commit inside the PR before squashing & merging.

It's weird because `Allow edits from maintainers` option is enabled!

**Comment by Mexidense** [comment]
I've disabled and enabled again... :shamrock:

**Comment by tristanjahier** [comment]
@Mexidense It works! It is not very practical since I need to clone your repository but it finally worked.
https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork

**Pull request merged**
